sycl/doc/extensions/SPIRV/SPV_INTEL_float_controls2.asciidoc

```diff
@@ -71,62 +74,123 @@ To use this extension within a SPIR-V module, the following
 
 _Target Width_ is an unsigned 32-bit integer.
-| *RoundToInfinityINTEL* | <<Literal, 'Literal'>>  _Target Width_
+| <<Literal, 'Literal'>>  _Target Width_ | *RoundToInfinityINTEL*
 | 5621 | *RoundingModeRTNINTEL* +
 The default rounding mode for floating-point arithmetic and conversions instructions must be round to negative infinity.
 If an instruction is decorated with *FPRoundingMode* or defines a rounding mode in its description, that rounding mode is applied and *RoundingModeRTNINTEL* is ignored.
 Only affects instructions operating on a floating-point type whose component width is _Target Width_.
 
 _Target Width_ is an unsigned 32-bit integer.
-| *RoundToInfinityINTEL* | <<Literal, 'Literal'>> _Target Width_
-| 5622 | *FloatingPointModeALTINTEL* +
+| <<Literal, 'Literal'>> _Target Width_ | *RoundToInfinityINTEL*
+  | 5622 | *FloatingPointModeALTINTEL* +
 The default floating-point operation mode for floating-point arithmetic and conversions instructions must be the Alternative Floating-Point Mode as specified by the client API.
 Only affects instructions operating on a floating-point type whose component width is _Target Width_.
 
 _Target Width_ is an unsigned 32-bit integer.
-| *FloatingPointModeINTEL* | <<Literal, 'Literal'>> _Target Width_
+| <<Literal, 'Literal'>> _Target Width_ | *FloatingPointModeINTEL*
 | 5623 | *FloatingPointModeIEEEINTEL* +
 The default floating-point operation mode for floating-point arithmetic and conversions instructions must be the IEEE 754 Mode as specified by the client API.
 Only affects instructions operating on a floating-point type whose component width is _Target Width_.
 
 _Target Width_ is an unsigned 32-bit integer.
-| *FloatingPointModeINTEL* | <<Literal, 'Literal'>> _Target Width_
+| <<Literal, 'Literal'>> _Target Width_ | *FloatingPointModeINTEL*
 |====
 --
+Modify Section 3 Binary form, add new sub-sections after 3.16 FP Rounding Mode:
+--
+[[FP_Denorm_Mode]]*3.17. FP Denorm Mode*
+
+Denormalized values handling mode.
+
+[cols="^.^1,15,15",options="header",width = "50%"]
+|====
+2+^.^| FP Denorm Mode | Enabling Capabilities
+| 0x0 | *Preserve* +
+Denormalized values must be preserved. |
+| 0x1 | *FlushToZero* +
+Denormalized values must be flushed to zero. |
+|====
+[[FP_Operation_Mode]]*3.18. FP Operation Mode*
+
+Floating-point operation mode.
+
+[cols="^.^1,15,15",options="header",width = "50%"]
+|====
+2+^.^| FP Operation Mode | Enabling Capabilities
+| 0x0 | *IEEE* +
+Floating-point operation mode is IEEE 754 Mode. |
+| 0x1 | *ALT* +
+Floating-point operation mode is Alternative Mode. |
+|====
+--
+
+Modify Section 3.20, Decoration, add the following rows to the Decoration table:
+
+--
+[cols="1,20,5,5,10",options="header",width = "80%"]
+|====
+  2+^| Decoration  2+| Extra Operands | Enabling Capabilities
+| 5822 | *FunctionRoundingModeINTEL* +
+Apply to a function to overwrite the default rounding mode for all floating-point arithmetic and conversion instructions inside the function.
```

Comment:
I think that description of new execution modes like `RoundingModeRTPINTEL` should be updated as well - it explicitly states that `FPRoundingMode` rounding mode overrides global rounding mode, but we probably need to document the relationships between:
- `RoundingModeRTPINTEL` and `FunctionRoundingModeINTEL`, i.e. between execution mode and function-wide mode
- `FunctionRoundingModeINTEL` and `FPRoundingMode`, i.e. between function-wide and per-instruction mode

Reply:
I added new rules to describe the interactions you mentioned.